test/curve/ankr.js

```diff
@@ -0,0 +1,110 @@
+describe("Mainnet Curve Ankr", function() {
+  let accounts;
+
+  // external contracts
+  let underlying;
+
+  // external setup
+  let underlyingWhale = "0x2FedC4002A2398ac32d9d43a9B92A5cf16a36304";
+
+  // parties in the protocol
+  let governance;
+  let farmer1;
+
+  // numbers used in tests
+  let farmerBalance;
+
+  // Core protocol contracts
+  let controller;
+  let vault;
+  let strategy;
+
+  let crv = "0xD533a949740bb3306d119CC777fa900bA034cd52";
+  let onx = "0xE0aD1806Fd3E7edF6FF52Fdb822432e847411033";
+  let ankr = "0x8290333ceF9e6D528dD5618Fb97a76f268f3EDD4";
+  let weth = "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2";
+
+  async function setupExternalContracts() {
+    underlying = await IERC20.at("0xaA17A236F2bAdc98DDc0Cf999AbB47D47Fc0A6Cf");
+    console.log("Fetching Underlying at: ", underlying.address);
+  }
+
+  async function setupBalance(){
+    let etherGiver = accounts[9];
+    // Give whale some ether to make sure the following actions are good
+    await send.ether(etherGiver, underlyingWhale, "1" + "000000000000000000");
+
+    farmerBalance = await underlying.balanceOf(underlyingWhale);
+    await underlying.transfer(farmer1, farmerBalance, { from: underlyingWhale });
+  }
+
+  before(async function() {
+    governance = "0xf00dD244228F51547f0563e60bCa65a30FBF5f7f";
+    accounts = await web3.eth.getAccounts();
+
+    farmer1 = accounts[1];
+
+    // impersonate accounts
+    await impersonates([governance, underlyingWhale]);
+
+    await setupExternalContracts();
+    [controller, vault, strategy] = await setupCoreProtocol({
+      "existingVaultAddress": null,
+      "strategyArtifact": CRVStrategyAnkrETHMainnet,
+      "liquidation": {"onx": {"sushi": [onx, weth]},
+                      "crv": {"uni": [crv, weth]},
+                      "ankr": {"uni": [ankr, weth]}},
+      "underlying": underlying,
+      "governance": governance,
+    });
+
+    // whale send underlying to farmers
+    await setupBalance();
+  });
+
+  describe("Happy path", function() {
+    it("Farmer should earn money", async function() {
+      let farmerOldBalance = new BigNumber(await underlying.balanceOf(farmer1));
+      await depositVault(farmer1, underlying, vault, farmerBalance);
+
+      // Using half days is to simulate how we doHardwork in the real world
+      let hours = 10;
+      let oldSharePrice;
+      let newSharePrice;
+      for (let i = 0; i < hours; i++) {
+        console.log("loop ", i);
+        let blocksPerHour = 240;
+        oldSharePrice = new BigNumber(await vault.getPricePerFullShare());
+        await controller.doHardWork(vault.address, { from: governance });
+        newSharePrice = new BigNumber(await vault.getPricePerFullShare());
+
+        console.log("old shareprice: ", oldSharePrice.toFixed());
+        console.log("new shareprice: ", newSharePrice.toFixed());
+        console.log("growth: ", (newSharePrice.dividedBy(oldSharePrice)).toFixed());
```

Comment:
Some of our tests actually do this incorrectly (the growth is always shown as 1): a better way would be just dividing `newSharePrice.toFixed() / oldSharePrice.toFixed()`

Reply:
I will change it.